src/tainting/Taint_lval_env.ml

```diff
@@ -79,6 +79,8 @@ let normalize_lval lval =
   let* base, rev_offset =
     match base with
     | Var _ -> Some (base, rev_offset)
+    (* explicit dereference of `this` e.g. `this->x` *)
+    | Mem { e = Fetch { base = VarSpecial (This, _); rev_offset = [] }; _ }
```

Comment:
Shouldn't we handle more than just `this` here? Then we can write tests for `ptr->a` etc as suggested in a previous comment.

Reply:
Done